cognite/client/utils/_concurrency.py

```diff
@@ -275,28 +275,35 @@ def execute_tasks(
     """
     Will use a default executor if one is not passed explicitly. The default executor type uses a thread pool but can
     be changed using ConcurrencySettings.executor_type.
+
+    Results are returned in the same order as that given by tasks.
     """
     if ConcurrencySettings.uses_mainthread() or isinstance(executor, MainThreadExecutor):
         return execute_tasks_serially(func, tasks, fail_fast)
 
     executor = executor or ConcurrencySettings.get_thread_pool_executor(max_workers)
+    task_order = [id(task) for task in tasks]
+    assert not find_duplicates(task_order)  # this would mean the exact same object is passed > 1 times
```

Comment:
Will the assert give a good user feedback if it fails.

Reply:
Its not for the user

Reply:
I'm inclined to remove it

Reply:
Removed it